### PR TITLE
[WIP] RGPD Employee classe

### PR DESCRIPTION
**Pull request by ghost**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.1.x
| Description?  | Enable unsubscribe for employees and change http by https 
| Type?         | improvement
| Category?     | BO

Normaly an employes can unsubscribe from this list.
It's just a suggestion because we don't know the method on : http://www.prestashop.com/ajax/controller.php
I think you can send an e-mail to confirm it's the owner of the e-mail.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9135)

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
Hi @okom3pom, this is not a bug fix but 2 improvements you are suggesting here.

I'll let @eternoendless take a look at your pull request but you'd better do it on develop branch, isn't it?

Also, shouldnt this pr be completed by a function to unsubscribe employee from newsletter?

**Comment by ghost** [comment]
@mickaelandrieu 

I can't know the methods you use internally.
It is a suggestion to make Prestashop RGPD compliant.

I just check 1.7 dev and it's the same http is use not https and no method to unsubscribe.

I can put new PR in 1.7 dev but we also need this for 1.6

File: classes/Employee.php [hunk 1]
```diff
@@ -233,16 +233,26 @@ protected function updateTextDirection()
...
+                    'email' => $this->email,
+                    'method' => 'addMemberToNewsletter',
+                    'language' => $language->iso_code,
+                    'visitorType' => 1,
+                    'source' => 'backoffice'
+                ));
+            } else {
+                $params = http_build_query(array(
+                    'email' => $this->email,
+                    'method' => 'removeMemberToNewsletter', // We don't know the method
```
Comment (ghost) [inline classes/Employee.php:249]: Only prestashop know the good param['method'] and he probably not exist

Comment (mickaelandrieu) [reply classes/Employee.php:249]: I'm not aware of method to remove member form news letter, sounds like complicated to accept the contribution without this method.

Ping @Quetzacoalt91

Comment (Quetzacoalt91) [reply classes/Employee.php:249]: I don't know either, I have asked internally.

Comment (Quetzacoalt91) [reply classes/Employee.php:249]: Unfortunately, there is currently no such API endpoint for un-subscription.
Merchants can opt-out with the emails they receive.

Comment (ghost) [reply classes/Employee.php:249]: Ok, problem it's not only merchants it's also my employees.

But i undunderstand it's hard to make it.

I will close this PR un put a new one only with https.

You are informed :)

**Pull request closed without merging**